EIPS/eip-5792.md

```diff
@@ -1,390 +1,269 @@
 
-```json
-null
-```
+### Call Execution Atomicity
 
-## Rationale
+The `wallet_sendCalls` method accepts an array of `calls`. However, this proposal does not require that these calls are executed as part of a single transaction. This is so that EOA wallets can accept multiple calls too. That said, we expect that app developers might want to only submit batched calls if and only if they are executed atomically. This would reduce the number of error cases an app developer would need to handle.
+
+We initially proposed that multiple calls must be executed atomically, but after some debate we ultimately decided this was too opinionated. Instead, we are including a specification for an `atomicBatch` capability. This allows for EOA wallets to accept multiple calls and still gives developers the option to only submit batched calls if they are executed atomically.
 
-Account abstracted wallets, either via [EIP-3074](./eip-3074.md) or [EIP-4337](./eip-4337.md) or other specifications,
-have more capabilities than regular EOA accounts.
+### Call Gas Limit
 
-The `eth_sendTransaction` and `eth_getTransactionReceipt` methods limit the quality of in-dapp transaction
-construction and status tracking. It's possible for dapps to stop tracking transactions altogether and
-leave that UX to the wallet, but it is a better UX when dapps provide updates on transactions constructed within the
-dapp, because dapps will always have more context than the wallets on the action that a set of calls is
-meant to perform. For example, an approve and swap might both be related to a trade that a user is attempting to make.
-Without these APIs, it's necessary for a dapp to represent those actions as individual transactions.
+Our initial proposal included an optional `gas` field for each call in the `calls` field accepted by the `walletSendCalls` method. However, we realized this could be misleading because in the case of [ERC-4337](./eip-4337.md) wallets you cannot specify a gas limit per call, only a single gas limit for all calls in the user operation. We then proposed a single `gas` value that would apply to all of the calls. This works for [ERC-4337](./eip-4337.md) wallets, but not for EOA wallets. When we decided that EOA wallets should be able to handle multiple calls, we decided to just remove the `gas` field altogether.
 
 ## Backwards Compatibility
 
-Wallets that do not support the following methods should return error responses to the new JSON-RPC methods.
-Dapps MAY attempt to send the same bundle of calls via `eth_sendTransaction` when they receive a not implemented
-call, or otherwise indicate to the user that their wallet is not supported.
+Wallets that do not support the following methods should return error responses to the new JSON-RPC methods. Apps MAY attempt to send the same batch of calls via `eth_sendTransaction` when they receive a not implemented call, or otherwise indicate to the user that their wallet is not supported.
```

Comment:
In your proposal an app needs to call `wallet_getCapabilities` first. And handling errors if it's not supported. That's an extra roundtrip and odd API (error handling vs. explicit).

Following my suggestion above that would be saved.

Reply:
Can you make clear your suggestion: what call will the app make to the wallet in order to discover capabilities?

Reply:
FWIW I am open to the idea that `eth_requestAccounts` as the implicit connection method should possibly be replaced by some like `wallet_connect` RPC that shares accounts + capabilities + other info. Don't necessarily want to block this EIP on getting to clarity there, but if we can align on something better quickly we should!

Reply:
Yes I saw you commented on https://github.com/arein/dapp-smart-account/pull/1/files#diff-2ce6a27d54a5083f40751babba9d77c5529f0515cd5f8dc31a379406d52ff6cdR38 and I linked the reuest there as you asked.

Just making very clear that this has nothing to do with _WalletConnect_. _WalletConnect_ has many things, one of them is a transport layer for remote wallets.

Our SDKs use the transport layer and support CAIPs first class. But many browser extension wallets use CAIPs without using our transport layer (even though some do). Our SDKs will also support other transport layers.

Reply:
Thanks. So is the rpc provider_authorize?